app/src/main/java/ch/epfl/sdp/blindly/main_screen/MatchPageFragment.kt

```diff
@@ -59,18 +72,81 @@ class MatchPageFragment : Fragment() {
         // Inflate the layout for this fragment
         val view = inflater.inflate(R.layout.fragment_match_page, container, false)
 
-        val intent = Intent(context, MatchActivity::class.java)
-        val profiles = createProfiles()
-        intent.putParcelableArrayListExtra(
-            "EXTRA_MATCH_PROFILES",
-            profiles as ArrayList<out Parcelable>
-        )
-        val matchActivityButton = view.findViewById<Button>(R.id.button_to_match)
-        matchActivityButton.setOnClickListener { startActivity(intent) }
+        setupButtons(view)
+        setupCardStackView(view)
 
         return view
     }
 
+    /**
+     * Do some action when the card is dragged (still holded)
+     *
+     * @param direction where the card is dragged (left, right)
+     * @param ratio between default (0) and swiped (1)
+     */
+    override fun onCardDragging(direction: Direction, ratio: Float) {
+    }
+
+    /**
+     * Do some actions when the card is swiped
+     *
+     * @param direction the direction the card is swiped (left, right)
+     */
+    override fun onCardSwiped(direction: Direction) {
+    }
+
+    override fun onCardRewound() {
+    }
+
+    override fun onCardCanceled() {
+    }
+
+    /**
+     * Do some action when the card appear
+     *
+     * @param view in which the card is
+     * @param position in the view
+     */
+    override fun onCardAppeared(view: View, position: Int) {
+    }
+
+    /**
+     * Do some action when the card disappears
+     *
+     * @param view in which the card was
+     * @param position in the view
+     */
+    override fun onCardDisappeared(view: View, position: Int) {
+    }
+
+    /**
+     * Initialize the manager, the adapter and the cardStackView
+     *
+     */
+    private fun setupCardStackView(view: View) {
+        cardStackView = view.findViewById(R.id.card_stack_view)!!
+        manager = CardStackLayoutManager(context, this)
+        manager.setStackFrom(StackFrom.None)
```

Comment:
Could all these calls to set() not be chained?

Reply:
Unfortunately no, because those methods don't return a CardStackLayoutManager so you can't link them :(